app/controllers/kiosk_reservations_controller.rb

```diff
@@ -102,7 +102,11 @@ def respond_error(message)
   def check_kiosk_enabled
     return if current_facility&.kiosk_enabled? && SettingsHelper.feature_on?(:kiosk_view)
 
-    raise NUCore::PermissionDenied
+    if current_user
+      render "errors/forbidden", status: :forbidden
+    else
+      return redirect_to new_user_session_path, alert: "You need to login to access this page."
+    end
```

Comment:
is this change necessary? It's not clear to me why if there's a current user an error is renderd and if not a redirection to login is done